lxd/db.go

```diff
@@ -402,14 +403,21 @@ func dbImageGet(d *Daemon, name string, public bool) (*shared.ImageBaseInfo, err
 	// count potential images first, if more than one
 	// return error
 	var countImg int
+	var err error
 	q := "SELECT count(id) FROM images WHERE fingerprint like ?"
 	if public {
 		q = q + " AND public=1"
 	}
-	err := d.db.QueryRow(q, (name + "%")).Scan(&countImg)
 
-	if err != nil {
-		return nil, err
+	for {
+		err = d.db.QueryRow(q, (name + "%")).Scan(&countImg)
+		if err == nil {
+			break
+		} else if ! shared.IsDbLockedError(err) {
+			shared.Debugf("dbImageGet: search for count of image by fp return error: %q\n", err)
+			return nil, err
+		}
+		time.Sleep(100 * time.Millisecond)
```

Comment:
I think we ought to have a timeout at some point, in the unlikely event that we've deadlocked ourselves.

Can we return an error after a 30s wait?